libraries/physics/src/CharacterController.cpp

```diff
@@ -543,12 +543,8 @@ void CharacterController::setLocalBoundingBox(const glm::vec3& minCorner, const
         _minStepHeight = DEFAULT_MIN_STEP_HEIGHT_FACTOR * (_halfHeight + _radius);
         _maxStepHeight = DEFAULT_MAX_STEP_HEIGHT_FACTOR * (_halfHeight + _radius);
 
-        if (_physicsEngine) {
```

Comment:
The issue was that on startup, before the physics engine loaded, we would set the ADD flags repeatedly as we scaled to the correct size without setting the REMOVE flags, so we'd end up with a ton of duplicate motion states.  The number would depend on loading times and the target scale (smaller scales take longer to get to, so the problem is worse).  In one case, I had 1500 motion states when it should have been 50.

Reply:
Hmm, what could be the original reasoning for the `if (_physicsEngine)`? Is that an attempt at optimization, or could it be protecting something?

Reply:
my guess is that instead of setting flags, this code used to do actual work here.  I don’t see any way that removing this if could cause problems.  perhaps if we actually added an if around the ADD flags, we could optimize this to do less work on startup, but that seems a lot riskier, since I could imagine it leading to cases where we never create the motion states